classes/class-wc-connect-taxjar-integration.php

```diff
@@ -8,80 +8,783 @@ class WC_Connect_TaxJar_Integration {
+	 * See: https://developers.taxjar.com/api/reference/#countries
+	 *
+	 * @return array Countries supported by TaxJar.
+	 */
+	public function get_supported_countries() {
+		$supported_countries = array_merge(
+			array(
+				'US',
+				'CA',
+				'AU',
+			),
+			WC()->countries->get_european_union_countries()
+		);
+
+		return $supported_countries;
+	}
+
+	/**
+	 * Check if a given country is supported by TaxJar.
+	 *
+	 * @param $country Two character country code.
+	 *
+	 * @return bool Whether or not the country is supported by TaxJar.
+	 */
+	public function is_supported_country( $country ) {
+		return in_array( $country, $this->get_supported_countries() );
+	}
+
+	/**
+	 * Gets the store's location settings.
+	 *
+	 * Modified version of TaxJar's plugin.
+	 * See: https://github.com/taxjar/taxjar-woocommerce-plugin/blob/82bf7c58/includes/class-wc-taxjar-integration.php#L796
+	 *
+	 * @return array
+	 */
+	public function get_store_settings() {
+		$store_settings = array(
+			'taxjar_zip_code_setting' => WC()->countries->get_base_postcode(),
+			'store_state_setting'     => WC()->countries->get_base_state(),
+			'store_country_setting'   => WC()->countries->get_base_country(),
+			'taxjar_city_setting'     => WC()->countries->get_base_city(),
+		);
+
+		return $store_settings;
+	}
+
+	/**
+	 * @param $message
+	 */
+	public function _log( $message ) {
+		$formatted_message = is_scalar( $message ) ? $message : json_encode( $message );
+
+		$this->logger->debug( $formatted_message, 'WCS Tax' );
+	}
+
+	/**
+	 * Calculate tax / totals using TaxJar at checkout
+	 *
+	 * Unchanged from the TaxJar plugin.
+	 * See: https://github.com/taxjar/taxjar-woocommerce-plugin/blob/82bf7c587/includes/class-wc-taxjar-integration.php#L475
+	 *
+	 * @return void
+	 */
+	public function calculate_totals( $wc_cart_object ) {
+		global $woocommerce;
+
+		// Get all of the required customer params
+		$taxable_address = $woocommerce->customer->get_taxable_address(); // returns unassociated array
+		$taxable_address = is_array( $taxable_address ) ? $taxable_address : array();
+
+		$to_country = isset( $taxable_address[0] ) && ! empty( $taxable_address[0] ) ? $taxable_address[0] : false;
+		$to_state = isset( $taxable_address[1] ) && ! empty( $taxable_address[1] ) ? $taxable_address[1] : false;
+		$to_zip = isset( $taxable_address[2] ) && ! empty( $taxable_address[2] ) ? $taxable_address[2] : false;
+		$to_city = isset( $taxable_address[3] ) && ! empty( $taxable_address[3] ) ? $taxable_address[3] : false;
+		$line_items = array();
+		$cart_taxes = array();
+
+		foreach ( $wc_cart_object->get_cart() as $cart_item_key => $cart_item ) {
+			$product = $cart_item['data'];
+			$id = $product->get_id();
+			$quantity = $cart_item['quantity'];
+			$unit_price = $product->get_price();
+			$line_subtotal = $cart_item['line_subtotal'];
+			$discount = ( $unit_price - $wc_cart_object->get_discounted_price( $cart_item, $unit_price ) ) * $quantity;
```

Comment:
`WC_Cart::get_discounted_price` is deprecated in WC `3.2`, and using it 💥 the checkout page. Works fine in 3.1

Reply:
Fixed in c402afd.